internal/nginx-meshctl/commands/services.go

```diff
@@ -96,3 +105,49 @@ func GetServices() *cobra.Command {
 
 	return cmd
 }
+
+// getEndpoints returns a slice of upstream addresses for a service.
+func getEndpoints(ctx context.Context, k8sClient client.Client, svc v1.Service) ([]string, error) {
+	endpointSlices := &discoveryv1beta1.EndpointSliceList{}
+	var upstreamAdresses []string
+	opt := client.MatchingLabels{"kubernetes.io/service-name": svc.Name}
+	if err := k8sClient.List(ctx, endpointSlices, opt); err != nil {
+		fmt.Printf("error getting list of endpoint slices for service: %v\n", err)
+		return nil, err
+	}
+	for _, epSlice := range endpointSlices.Items {
+		epSlice := epSlice
+		if epSlice.Namespace == svc.Namespace {
+			for _, endpoint := range epSlice.Endpoints {
+				upstreamAdresses = append(upstreamAdresses, endpoint.Addresses...)
+			}
+			return upstreamAdresses, nil
+		}
+	}
+	// in the case that a service has no upstreams yet but is in a namespace where injection would be enabled
+	// return nothing rather than an error since that service is still 'part' of the mesh.
+	return nil, nil
+}
+
+// isNamespaceInjectionEnabled returns whether a given namespace has injection enabled.
+func isNamespaceInjectionEnabled(ctx context.Context, k8sClient client.Client, ns string) (bool, error) {
+	meshClient, err := mesh.NewMeshClient(initK8sClient.Config(), meshTimeout)
+	if err != nil {
+		return false, fmt.Errorf("failed to get mesh client: %w", err)
+	}
+	meshConfig, err := GetMeshConfig(meshClient)
+	if err != nil {
+		return false, fmt.Errorf("unable to get mesh config: %w", err)
+	}
```

Comment:
This could be worth pulling out into it's own function

Reply:
FWIW, this is gonna go away with my API migration work and be replaced.